relay-server/src/metrics_extraction/event.rs

```diff
@@ -87,7 +135,11 @@ fn extract_event_tags(event: &Event, tags: &[TagSpec]) -> BTreeMap<String, Strin
     map
 }
 
-fn read_metric_value(event: &Event, field: Option<&str>, ty: MetricType) -> Option<MetricValue> {
+fn read_metric_value(
+    event: &impl FieldValueProvider,
+    field: Option<&str>,
+    ty: MetricType,
+) -> Option<MetricValue> {
```

Comment:
Now that these functions are independent of `Event`, we should move them to a different module. I'll keep them here for now to keep the diff readable.